doc/md/fundamentals/1-basic-syntax/9-operators.md

```diff
@@ -0,0 +1,152 @@
+
+Binary operators combine two numbers to produce a result.
+
+| Operator | Description |  Example|
+|----------|------------|------------|
+| `+`      | Addition | `a + b` |
+| `-`      | Subtraction | `a - b` |
+| `*`      | Multiplication | `a * b` |
+| `/`      | Division (integer division) | `a / b` |
+| `%`      | Modulus (remainder) | `a % b` |
+| `**`     | Exponentiation | `a ** b` |
+
+:::warning
+
+Division (`/`) on integers **truncates** decimals. For floating-point division, use `Float.fromInt()`:
+
+```motoko no-repl
+let result = Float.fromInt(10) / Float.fromInt(3);
+```
+
+:::
+
+## Bitwise operators
+
+Bitwise operators manipulate numbers **at the binary level**.
+
+| Operator | Description |Example |
+|----------|------------|----------|
+| `&`      | Bitwise AND |`a & b` |
+|  `|`    | Bitwise OR | `a | b` |
+| `^`      | Bitwise XOR | `a ^ b` |
+| `<<`     | Shift left | `a << b` |
+| `>>`     | Shift right (must be proceeded by whitespace) |`a >> b` |
+| `<<>`    | Rotate left (circular shift) | `a <<> b` |
+| `<>`     | Rotate right (circular shift)| `a <> b` |
+
+:::caution
+
+Bitwise operators can only be used with bounded types.
+
+:::
+
+## Wrapping operators
+
+Bounded integers **trap** on overflow, but **wrapping versions** allow overflow behavior.
+
+| Operator | Description | Example |
+|----------|------------|------------|
+| `+%`     | Addition with wrap-around | `a +% b` |
+| `-%`     | Subtraction with wrap-around | `a -% b` |
+| `*%`     | Multiplication with wrap-around | `a *% b` |
+| `**%`    | Exponentiation with wrap-around | `a **% b` |
+
+## Text operators
+
+| Operator | Description | Example |
+|----------|------------|------------|
+| `#`      | Concatenates two [`Text`](https://internetcomputer.org/docs/motoko/base/Text) values | `a # b` |
+
+## Assignment operators
+
+Assignment operators modify variables in place. Only mutable variables `var` can be reassigned.
+
+| Operator | Description |Examples|
+|----------|------------|---------|
+| `:=`     | Assign a value | `a := b` |
+| `+=`     | Add and assign | `a += b` |
+| `-=`     | Subtract and assign | `a -= b` |
+| `*=`     | Multiply and assign | `a *= b` |
+| `/=`     | Divide and assign | `a /= b` |
+| `#=`     | Concatenate and assign (for [`Text`](https://internetcomputer.org/docs/motoko/base/Text)) | `a #= b` |
+
+## Operator precedence
+
+Operators follow precedence rules, meaning some operators are evaluated before others.
+
+1. Unary operators (`-`, `!`, `^`)
+2. Exponentiation (`**`, `**%`)
+3. Multiplication & division (`*`, `/`, `%`, `*%`)
+4. Addition & subtraction (`+`, `-`, `+%`, `-%`)
+5. Bitwise operators (`&`, `|`, `^`)
+6. Comparison operators (`==`, `!=`, `<`, `>`, `<=`, `>=`)
+7. Assignment operators (`:=`, `+=`, `-=`, etc.)
+
+For example:
+
+```motoko
+let result = 10 + 5 * 2; // result = 20
+```
+
+Use parentheses to enforce a different order.
+
+```motoko
+let result = (10 + 5) * 2; // result = 30
+```
+
+## Pipes
```

Comment:
Would a section on `do ? { ... option! ... }` fit here with other operators? This is a like an unary operator